libraries/integration/Microsoft.Bot.Builder.Integration.NetCore/BotFrameworkOptions.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.AspNetCore.Http;
+using BotMiddleware = Microsoft.Bot.Builder.Middleware;
+using System.Collections.Generic;
+
+namespace Microsoft.Bot.Builder.Integration.NetCore
+{
+    public class BotFrameworkOptions
+    {
+        private readonly List<BotMiddleware.IMiddleware> _middleware;
+
+        public BotFrameworkOptions()
+        {
+            _middleware = new List<BotMiddleware.IMiddleware>();
+
+            RouteBaseUrl = "/bot";
+        }
+
+        public PathString RouteBaseUrl { get; set; }
+        public string ApplicationId { get; set; }
+        public string ApplicationPassword { get; set; }
+        public List<BotMiddleware.IMiddleware> Middleware { get => _middleware; }
```

Comment:
Why not have this registered in the main Container as middleware? This way we won't be needing the options class at all (Creds from ICredentialProvider and Middlewares from MiddlewareSet)